src/domain/repositories/UserMonitoringMetadataRepository.ts

```diff
@@ -0,0 +1,68 @@
+import { ProgramMetadata, UserRes, UserRoleAuthority, User } from "data/d2-users/D2Users.types";
+import { Async } from "domain/entities/Async";
+
+export interface UserMonitoringMetadataRepository {
+    getTemplateAuthorities(options: UsersOptions): Promise<Async<TemplateGroupWithAuthorities[]>>;
+    getMetadata(programId: string): Promise<Async<ProgramMetadata>>;
+    getAllUsers(excludeIds: string[], exclude?: boolean): Promise<Async<User[]>>;
+}
+
+export interface UserMonitoringDetails extends UserMonitoringCountResponse {
```

Comment:
There are some interfaces here that are the same in `UserMonitoring.ts`, can we reuse them?